src/qilib/utils/serialization.py

```diff
@@ -189,7 +189,7 @@ def register_dataclass(self, type_: type) -> None:
             type_: The dataclass type to decode
         """
 
-        type_name = type_.__name__
+        type_name = f'dataclass_{type_.__name__}'
```

Comment:
```suggestion
        type_name = f'_dataclass_{type_.__name__}'
```
I see now that the ticket specifies `dataclass_xxx`, but adding an underscore is better I think. @qSaevar